src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -234,6 +252,213 @@ protected ActionResult execute() throws EntityDoesNotExistException {
+                                 + feedbackResponseComment.createdAt + "<br>"
+                                 + "comment text: " + feedbackResponseComment.commentText.getValue();
+    }
+
+    // Extracts two types of feedback response comments- ones which were newly added and ones which were edited
+    private void extractFeedbackResponseCommentsForResponseData(FeedbackQuestionAttributes questionAttributes,
+                                                                int questionIndex, FeedbackResponseAttributes response,
+                                                                int responseIndex) {
+        if (!questionAttributes.getQuestionDetails().isStudentCommentsOnResponsesAllowed()) {
+            return;
+        }
+        //When comment is added, comment text parameter is of type responsecommenttext-responseIndex-1-questionIndex
+        String commentText = getRequestParamValue(Const.ParamsNames.FEEDBACK_RESPONSE_COMMENT_TEXT
+                                                          + "-" + responseIndex + "-"
+                                                          + Const.GIVER_INDEX_FOR_FEEDBACK_SUBMISSION_PAGE
+                                                          + "-" + questionIndex);
+        if (!StringHelper.isEmpty(commentText)) {
+            String commentIndex = "-" + responseIndex + "-"
+                                          + Const.GIVER_INDEX_FOR_FEEDBACK_SUBMISSION_PAGE + "-" + questionIndex;
+
+            questionIdsForComments.put(commentIndex, questionAttributes.getId());
+            responseGiverMapForComments.put(commentIndex, response.giver);
+            responseRecipientMapForComments.put(commentIndex, response.recipient);
+            commentsToAddText.put(commentIndex, commentText);
+            return;
+        }
+
+        //commentId exists only when comment is edited but not when it is newly added
+        // and it is of type responsecommentid-responseIndex-questionIndex-1
+        String commentId =
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_RESPONSE_COMMENT_ID
+                                             + "-" + responseIndex + "-"
+                                             + Const.GIVER_INDEX_FOR_FEEDBACK_SUBMISSION_PAGE
+                                             + "-" + questionIndex + "-" + 1);
+        String editedCommentText =
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_RESPONSE_COMMENT_TEXT
+                                             + "-" + responseIndex + "-"
+                                             + Const.GIVER_INDEX_FOR_FEEDBACK_SUBMISSION_PAGE
+                                             + "-" + questionIndex + "-" + 1);
+        if (commentId != null && response.getId() != null) {
+            FeedbackResponseCommentAttributes commentCheck =
+                    logic.getFeedbackResponseComment(Long.parseLong(commentId));
+
+            if (!StringHelper.isEmpty(editedCommentText)
```

Comment:
No need to do the check as we are totally fine to write the same data to the database. A more fancy way would be to introduce the dirty flag (which is more complicated). Let's make things simple.